src/atoms/projectState.ts

```diff
@@ -0,0 +1,82 @@
+import { Atom, atom } from 'jotai';
+import { selectAtom } from 'jotai/utils';
+
+import { getSeparator, newProject, openProject, sampleProject } from '../io/filesystem';
+import { closeAllEditorsAtom } from './editorActions';
+import { refreshFileTreeAtom } from './fileExplorerActions';
+import { clearAllReferencesAtom, loadReferencesAtom } from './referencesState';
+
+// #####################################################################################
+// Internal Atoms
+// #####################################################################################
+const currentProjectPathAtom = atom('');
+
+// #####################################################################################
+// Public API
+// #####################################################################################
+export const isProjectOpenAtom = atom((get) => get(currentProjectPathAtom) !== '');
+export const projectPathAtom: Atom<string> = currentProjectPathAtom;
+export const projectNameAtom = selectAtom(currentProjectPathAtom, extractProjectName);
```

Comment:
nit: I don't think we need `selectAtom` here, we can use a basic derived atom like this `atom((get) => extractProjectName(get(currentProjectPathAtom)));`

Reply:
I think it's ok and more legible.